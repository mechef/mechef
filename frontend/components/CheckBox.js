// @flow

import * as React from 'react';

import {
  primaryColor
} from '../utils/styleVariables';

type Props = {
  children: React.Node,
  checked: boolean,
  onChange: (isChecked: boolean) => mixed,
};

type State = {
  isChecked: boolean,
}

class CheckBox extends React.Component<Props, State> {

  static defaultProps = {
    children: '',
    checked: false,
    onChange: () => { },
  }

  constructor(props: Props) {
    super(props);
    this.state = {
      isChecked: props.checked,
    }
  }

  componentWillReceiveProps(newProps: Props) {
    this.setState({
      isChecked: newProps.checked,
    })
  }

  handleChange = () => {
    this.props.onChange(!this.state.isChecked);
    this.setState({
      isChecked: !this.state.isChecked,
    });
  }

  render() {
    return (
      <div className="checkboxContainer">
        <button type="button" onClick={this.handleChange} className={this.state.isChecked ? 'checkedIcon' : 'uncheckedIcon'} />
        <div className="checkboxLabel">{this.props.children}</div>
        <style jsx>
          {`
            .checkboxContainer {
              display: flex;
              justify-content: space-between;
              align-items: center;
              width: 116px;
            }

            button {
              outline: none;
              border: 0;
              border-radius: 4px;
              cursor: pointer;
            }

            .checkboxInput {
              display: none;
              background-color: ${primaryColor};
            }

            .checkedIcon {
              width: 20px;
              height: 20px;
              background-image: url('../static/svg/checkbox.svg');
            }

            .uncheckedIcon {
              width: 20px;
              height: 20px;
              background-image: url('../static/svg/checkbox_grey.svg');
            }

            .checkboxLabel {
              font-weight: 400;
              width: 84px;
            }
          `}
        </style>
      </div>
    );
  }
}

export default CheckBox;
