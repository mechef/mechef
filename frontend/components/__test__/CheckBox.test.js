import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';

import CheckBox from '../CheckBox';

describe('Pages With Snapshot Testing', () => {
  it('snapshot CheckBox Component', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<CheckBox onChange={isChecked => {}} checked />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
