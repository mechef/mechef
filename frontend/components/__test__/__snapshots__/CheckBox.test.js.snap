// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot CheckBox Component 1`] = `
<div
  className="jsx-606403950 checkboxContainer"
>
  <button
    className="jsx-606403950 checkedIcon"
    onClick={[Function]}
    type="button"
  />
  <div
    className="jsx-606403950 checkboxLabel"
  >
    
  </div>
  <JSXStyle
    css=".checkboxContainer.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;max-width:116px;width:100%;}button.__jsx-style-dynamic-selector{outline:none;border:0;border-radius:4px;cursor:pointer;}.checkboxInput.__jsx-style-dynamic-selector{display:none;background-color:#3e9f40;}.checkedIcon.__jsx-style-dynamic-selector{width:20px;height:20px;background-image:url('../static/svg/checkbox.svg');}.uncheckedIcon.__jsx-style-dynamic-selector{width:20px;height:20px;background-image:url(../static/svg/checkbox_grey.svg);}.checkboxLabel.__jsx-style-dynamic-selector{font-weight:400;max-width:84px;}"
    dynamic={
      Array [
        "#3e9f40",
      ]
    }
    styleId="3405802264"
  />
</div>
`;
